src/aggregator/client/conn.go

```diff
@@ -45,6 +50,27 @@ type sleepFn func(time.Duration)
 type connectWithLockFn func() error
 type writeWithLockFn func([]byte) error
 
+type connWriter interface {
+	io.Writer
+	Flush() error
+}
+
+type rawConnWriter struct {
+	writer io.Writer
+}
+
+func newRawConnWriter(w io.Writer) connWriter {
+	return &rawConnWriter{writer: w}
+}
+
+func (w *rawConnWriter) Write(p []byte) (int, error) {
+	return w.writer.Write(p)
+}
+
+func (w *rawConnWriter) Flush() error {
+	return nil
```

Comment:
What if the underlying writer is in some kind of an error state, for example, the connection is closed?